packages/application-components/src/components/modal-pages/internals/modal-page.tsx

```diff
@@ -0,0 +1,136 @@
+  getBeforeCloseOverlayAnimation,
+  getBeforeCloseContainerAnimation,
+} from './modal-page.styles';
+import ModalPageTopBar from './modal-page-top-bar';
+
+// When running tests, we don't render the AppShell. Instead we mock the
+// application context to make the data available to the application under
+// test. However, this also means that the AppShell tree is not rendered,
+// which contains among other things the <div id="portals-container" />
+// element, used to render the modal content.
+// Apparently though, when the <Modal> component unmounts between tests, it
+// tries to remove the child elements within the parent selector
+// ("portals-container") which may have been cleaned up by another test,
+// resulting in the <Modal> to throw an NotFoundError.
+// To solve this, we can simply use "document.body" as the parent selector
+// instead of a specific element that will be cleaned up, resulting in
+// console errors (even though the test passes). We only need to to this in
+// test environment.
+const getDefaultParentSelector = () =>
+  process.env.NODE_ENV === 'test'
+    ? document.body
+    : (document.querySelector<HTMLElement>(
+        `#${PORTALS_CONTAINER_ID}`
+      ) as HTMLElement);
+
+type Label = string | MessageDescriptor;
+type Props = {
+  level: number;
+  title: string;
+  isOpen: boolean;
+  onClose?: (event: React.SyntheticEvent) => void;
+  children: React.ReactNode;
+  zIndex?: number;
+  baseZIndex: number;
+  getParentSelector: typeof getDefaultParentSelector;
+  shouldDelayOnClose: boolean;
+  // TopBar props:
+  topBarColor?: 'surface' | 'neutral';
+  currentPathLabel?: string;
+  previousPathLabel: Label;
+};
+const defaultProps: Pick<
+  Props,
+  'level' | 'baseZIndex' | 'getParentSelector' | 'shouldDelayOnClose'
+> = {
+  level: 1,
+  baseZIndex: 1000,
+  getParentSelector: getDefaultParentSelector,
+  shouldDelayOnClose: true,
+};
+
+const ModalPage = (props: Props) => {
+  const [forceClose, setForceClose] = React.useState(false);
+  const closingTimer = React.useRef<NodeJS.Timeout>();
+  React.useEffect(() => {
+    if (props.isOpen === true) setForceClose(false);
+    return () => {
+      if (closingTimer.current) {
+        clearTimeout(closingTimer.current);
+      }
+    };
+  }, [props.isOpen]);
+  const { onClose } = props;
+  const handleClose = React.useCallback(
+    event => {
+      if (props.shouldDelayOnClose) {
+        // In this case we want the closing animation to be shown
+        // and therefore we need wait for it to be completed
+        // before calling `onClose`.
+        setForceClose(true);
+        closingTimer.current = setTimeout(() => {
+          onClose && onClose(event);
+        }, TRANSITION_DURATION);
+        return;
+      }
+      onClose && onClose(event);
+    },
+    [onClose, props.shouldDelayOnClose]
+  );
+  return (
+    <ClassNames>
+      {({ css: makeClassName }) => (
+        <Modal
+          isOpen={forceClose === true ? false : props.isOpen}
```

Comment:
Now that we `useState`, it was a bit weird before to use `null` and `false` and the state values. Now it's either `true` or `false`, the outcome should be the same though.